src/manifest.ts

```diff
@@ -1,31 +1,45 @@
 import { z } from "zod";
 
 // https://github.com/opencontainers/image-spec/blob/main/manifest.md
-export const manifestSchema = z.object({
-  schemaVersion: z.literal(2),
-  artifactType: z.string().optional(),
-  // to maintain retrocompatibility of the registry, let's not assume mediaTypes
-  mediaType: z.string(),
-  config: z.object({
+export const manifestSchema = z
+  .object({
+    schemaVersion: z.literal(2),
+    artifactType: z.string().optional(),
+    // to maintain retrocompatibility of the registry, let's not assume mediaTypes
     mediaType: z.string(),
-    digest: z.string(),
-    size: z.number().int(),
-  }),
-  layers: z.array(
-    z.object({
-      size: z.number().int(),
-      mediaType: z.string(),
-      digest: z.string(),
-    }),
-  ),
-  annotations: z.record(z.string()).optional(),
-  subject: z
-    .object({
+    config: z.object({
       mediaType: z.string(),
       digest: z.string(),
       size: z.number().int(),
-    })
-    .optional(),
-});
+    }),
+    layers: z.array(
+      z.object({
+        size: z.number().int(),
+        mediaType: z.string(),
+        digest: z.string(),
+      }),
+    ),
+    annotations: z.record(z.string()).optional(),
+    subject: z
+      .object({
+        mediaType: z.string(),
+        digest: z.string(),
+        size: z.number().int(),
+      })
+      .optional(),
+  })
+  .or(
+    z
+      .object({
+        schemaVersion: z.literal(1),
```

Comment:
this is lazy schema definition, but due to the wildness of schemaVersion 1 I am not brave enough to put more constraints here :)